### DISCUSSION TITLE
Support for strict concurrency

### DISCUSSION THREAD

**Discussion by roaring-b** [netreconlab/Parse-Swift#119] [Ideas]
Enforcing strict concurrency in Swift reveals that ParseACL does not (yet) conform to the Sendable protocol. It's possible adding that conformance to ParseACL will be all that's needed to get ParseSwift to support strict concurrency. For now, warnings can be disabled by adding `@preconcurrency` before each `import ParseSwift` but this does not fix the underlying issue. (Wasn't sure whether to make this an _issue_ without more discussion.)

**Comment by cbaker6** [comment]
`ParseACL` is a value type, are you experiencing any concurrency issue with `ParseACL` or any other part of the SDK? If not, the warnings can be ignored for now as it doesn't seem like there's any real issue with ParseSwift.

Making all types conform to `Sendable` and enforcing concurrency will not be looked at until Parse Swift version 6.0.0 (release slated for 2024) as it will require breaking changes. In 5.x, there should doesn't seem to be any concurrency issues, only warnings.

**Comment by roaring-b** [comment]
Thanks for the quick response. I think it is just the warnings. I just couldn't tell if there was a quick and easy win to be had there.

**Reply by cbaker6** [reply]
Unfortunately, there's no quick and easy way as: 

> Making all types conform to Sendable and enforcing concurrency will not be looked at until Parse Swift version 6.0.0 (release slated for 2024) as it will require breaking changes. In 5.x, there should doesn't seem to be any concurrency issues, only warnings.

The easiest way currently will be via the method you mentioned earlier:

> For now, warnings can be disabled by adding @preconcurrency before each import ParseSwift but this does not fix the underlying issue. (Wasn't sure whether to make this an issue without more discussion.)

**Reply by cbaker6** [reply]
See #174
